src/common/UsernameSet.cpp

```diff
@@ -4,6 +4,27 @@
 
 namespace chatterino {
 
+std::pair<UsernameSet::Iterator, bool> findOrErase(
+    std::set<QString, CaseInsensitiveLess> &set, const QString &value)
+{
+    if (!value.isLower())
+    {
+        auto iter = set.find(value);
+        if (iter != set.end())
+        {
+            if (QString::compare(*iter, value, Qt::CaseSensitive) != 0)
+            {
+                set.erase(iter);
+            }
+            else
+            {
+                return {iter, false};
+            }
+        }
+    }
+    return {set.end(), true};
+}
+
```

Comment:
Can we put this into the anonymous namespace? Like here for example:

https://github.com/Chatterino/chatterino2/blob/805e33e766a6019d26284851f5821fc00237751c/src/widgets/dialogs/switcher/QuickSwitcherPopup.cpp#L16-L31

Reply:
Sure